### PR TITLE
Issue #5508: mutation testing setup for research aggregation (cheap-lane worker)

### PR DESCRIPTION
**Pull request by ll7**
## What / Why

This PR implements the first slice of issue #5508:
- Selected `mutmut` as the mutation-testing tool for Python.
- Configured and executed mutation testing on `robot_sf/research/aggregation.py`, including the copied runtime package needed for the selected tests to run in mutmut's isolated tree.
- Hardened `tests/research/test_aggregation.py` with 9 additional test cases and stronger assertions for statistical metrics, confidence intervals, metadata exclusion, completeness ordering, and edge cases.
- Corrected numeric seed ordering in `compute_completeness_score`: numeric seed IDs now sort numerically before non-numeric IDs.
- Recorded the reproducible diagnostic result and survivor/no-test triage in `mutation_testing_triage.md`.

## Validation Output

- `uv run pytest tests/research/test_aggregation.py` — 16 passed.
- `uv run ruff check robot_sf/research/aggregation.py tests/research/test_aggregation.py` — passed.
- `uv run ruff format --check robot_sf/research/aggregation.py tests/research/test_aggregation.py` — passed.
- `uv run mutmut run --max-children 8` — 423 mutants generated: 200 killed, 5 equivalent survivors, 218 no-test mutants, and 0 timeout/suspicious results.

## Successor / What Remains

Refs #5508

Remaining tasks:
- Set up a scheduled diagnostic job (for example, a weekly/nightly GitHub Actions workflow) that runs the configured `mutmut` setup, produces a surviving-mutant report, and alerts on regressions.
- Add targeted tests for the 218 no-test mutants covering `export_metrics_json`, `export_metrics_csv`, `_load_manifest_payload`, and `extract_seed_metrics`; these functions remain outside this first slice's selected test file and are tracked by #5508.

This PR was produced by the agy/Gemini-3.5-Flash cheap implementation lane; the review gate hardened and validated it.

## Research Result Guidance

- Target/hypothesis: the selected aggregation tests detect meaningful mutations in the aggregation/completeness paths.
- Comparator/baseline: the current `robot_sf/research/aggregation.py` implementation and its selected unit tests.
- Evidence tier: diagnostic-only; this is not benchmark evidence or a paper-facing mutation score.
- Result classification: 200/205 executable mutants were killed; 5 were reviewed as equivalent under the current schema, while 218 mutants had no selected test coverage.
- Decision/stop rule: keep #5508 open for scheduled automation and no-test coverage; do not promote this run as a repository-wide mutation score.
- Synthesis/update target: `mutation_testing_triage.md` and #5508.

## Domain-Aware Approval

- Required for this PR: yes — the diagnostic result concerns benchmark-critical test adequacy and must stay bounded.
- Domains reviewed: benchmark-critical metric aggregation, mutation-result interpretation, and evidence claim boundaries.
- Status: approved for this diagnostic-only slice, subject to the configured CI and merge checklist.
- Approver/review source or waiver: gate review at the repaired head; no paper-facing or benchmark-performance claim is approved.
- Validity checklist:
  - Target claim/hypothesis: selected tests detect meaningful mutations in aggregation/completeness paths.
  - Comparator or split/evidence validity: current implementation versus the selected unit-test contract; no benchmark split or ranking comparison is made.
  - Fallback/degraded exclusions: no fallback or degraded execution is counted; no-test mutants are excluded from the executable-mutant ratio.
  - Claim boundary: diagnostic-only; not benchmark evidence, a repository-wide mutation score, or a paper-facing result.
  - Implementation integrity vs experimental validity: tests and mutation output establish implementation-test adequacy only, not experimental validity.

## Downstream Propagation

- Parent issue: #5508 remains open for scheduled automation and the no-test coverage gap.
- Claim map/benchmark report: NA — no benchmark or paper-facing claim is made.
- Durable diagnostic artifact: `mutation_testing_triage.md`.

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — review at `55c6741aa2337f82e1177166d3446740e485623d`

The gate reviewed the PR diff against current `origin/main` (`fe606c664f47bc2fa58213bcf1b8126a0624bb10`). CodeRabbit’s visible review result is rate-limited; that is advisory only. The open review-thread count is **1**.

1. **Open bot thread (medium, `tests/research/test_aggregation.py:225`, [thread](https://github.com/ll7/robot_sf_ll7/pull/5513#discussion_r3571424400))** — `compute_completeness_score` currently sorts numeric seed strings lexicographically because `_seed_sort_key` converts `int(value)` back to `str`; the test therefore codifies `10` before `2` despite its numeric-ordering docstring. **Verdict: fixed** — phase 2 will return the integer sort key and assert numeric order.
2. **Gate finding (scope/evidence)** — the PR body claims an executed and documented `mutation_testing_triage.md` survivor report, but that file and any durable mutation-run summary are absent from the diff. Issue #5508’s first-slice definition of done requires survivor triage, so the current claim/contract is not reproducible from the branch. **Verdict: fixed** — phase 2 will add a compact tracked triage report tied to the actual run, or correct the claim and contract before merge.

No finding is waived. The PR remains blocked from merge until phase 2 repairs are pushed, targeted/full validation is rerun, and the post-push thread/checklist snapshot is green.

File: tests/research/test_aggregation.py [hunk 1]
```diff
@@ -130,12 +137,98 @@ def test_aggregate_metrics_missing_values():
...
+    """Verify completeness score correctly sorts missing and failed seeds."""
+    completeness = compute_completeness_score(
+        expected_seeds=[10, 2, "abc", 1, "1a"],
+        completed_seeds=[1],
+        failed_seeds=[10, 2, "abc", "1a"],
+    )
+    # missing: empty list as all others are in failed or completed
+    assert completeness["missing_seeds"] == []
+    # failed seeds should sort numeric first: 10, 2, 1a, abc
+    assert completeness["failed_seeds"] == ["10", "2", "1a", "abc"]
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/research/test_aggregation.py]: Fixed in d16610d14: _seed_sort_key now returns (0, int(value)); the regression test asserts ["2", "10", "1a", "abc"].

**Comment by ll7** [comment]
## Gate delta — review at `bbbaeb2d1cff126061286f0554aa5439c940a206`

`gh pr update-branch` added only the current `main` synchronization merge (`d16610d14` + `fe606c664`); `git diff origin/main...HEAD` leaves the reviewed PR changes unchanged. Prior gate findings at `d16610d14` stand. The post-push/post-sync review-thread snapshot reports **0 open review threads**. Final local readiness proof remains valid for the unchanged PR diff; fresh CI is now required for this synchronized head.
